log/log.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"fmt"
+	"os"
+
+	"go.uber.org/zap"
+)
+
+var logger *zap.Logger
+
+var DevMode bool
+
```

Comment:
this file looks like reinventing logger. Why do we need it?

Reply:
you need to pass zap as an instance somehow... This basically creates a static facade in front of a logging singleton instance.

Another variant would be to use `context.Context` and bind a logger to it, like it is done in kubernetes client's `log.FromContext()` and `log.IntoContext()`.

None of those seems to be available from zap as is directly.

Because context is not used in the codebase, I opted for the static facade. We don't need any advanced contextual logging with MDC-like fields because the codebase is not that complex (at least for now), so I think that's fine.

That said, zap seems to be a popular logging choice and is used under the hood in kubernetes runtime as well, AFAIK.

Reply:
what problem you are trying to solve? Is this related https://github.com/uber-go/zap/blob/master/FAQ.md#why-include-package-global-loggers ?

Reply:
Huh, didn't notice `zap.L()`! :)